exchanges/exchange.go

```diff
@@ -1136,3 +1113,112 @@ func (e *Base) SetSaveTradeDataStatus(enabled bool) {
 		log.Debugf(log.Trade, "Set %v 'SaveTradeData' to %v", e.Name, enabled)
 	}
 }
+
+// NewEndpoints declares default and running URLs maps
+func (e *Base) NewEndpoints() *Endpoints {
+	return &Endpoints{
+		defaults: make(map[string]string),
+	}
+}
+
+// SetDefaultEndpoints declares and sets the default URLs map
+func (e *Endpoints) SetDefaultEndpoints(m map[URL]string) {
+	for k, v := range m {
+		e.defaults[k.String()] = v
+	}
+}
+
+// SetRunning populates running URLs map
+func (e *Endpoints) SetRunning(key, val string, overwrite bool) error {
+	e.Lock()
+	defer e.Unlock()
+	_, err := url.ParseRequestURI(val)
+	if err != nil {
+		return fmt.Errorf("invalid url: %v", err)
```

Comment:
To add a bit more on what I said here: https://github.com/thrasher-corp/gocryptotrader/pull/530#discussion_r554667559
What I think should happen is that you log the issue here, rather than return error. Because when I put in a dumb URL, the exchange is just not loaded when we could just continue using the default URL instead.

We can just have it so that the invalid URL is just not set. So by doing the following:
```
_, err = url.ParseRequestURI(val)
	if err != nil {
		log.Warnf(log.ExchangeSys,
			"Could not set custom URL for %v to %v. err: %v. Using default value of %v",
			key,
			val,
			err,
			e.defaults[key])
		return nil
	}
```

Then you get an output like:
```
[WARN] |  13/01/2021 13:49:36  | Could not set custom URL for RestSpotURL to hello MadCozBadd!. err: parse "hello MadCozBadd!": invalid URI for request. Using default value of https://sapi.binance.com
```